src/huggingface_hub/utils/_runtime.py

```diff
@@ -0,0 +1,144 @@
+# coding=utf-8
+# Copyright 2022-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Check presence of installed packages at runtime."""
+import sys
+from typing import Optional
+
+import packaging.version
+
+from .. import __version__
+
+
+_PY_VERSION: str = sys.version.split()[0].rstrip("+")
+
+if packaging.version.Version(_PY_VERSION) < packaging.version.Version("3.8.0"):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata
+
+
+_package_versions = {}
+
+_CANDIDATES = {
+    "torch": {"torch"},
+    "pydot": {"pydot"},
+    "graphviz": {"graphviz"},
+    "tensorflow": (
+        "tensorflow",
+        "tensorflow-cpu",
+        "tensorflow-gpu",
+        "tf-nightly",
+        "tf-nightly-cpu",
+        "tf-nightly-gpu",
+        "intel-tensorflow",
+        "intel-tensorflow-avx512",
+        "tensorflow-rocm",
+        "tensorflow-macos",
+    ),
+    "fastai": {"fastai"},
+    "fastcore": {"fastcore"},
+    "jinja": {"Jinja2"},
+}
+
+# Check once at runtime
+for candidate_name, package_names in _CANDIDATES.items():
+    _package_versions[candidate_name] = "N/A"
+    for name in package_names:
+        try:
+            _package_versions[candidate_name] = importlib_metadata.version(name)
+            break
+        except importlib_metadata.PackageNotFoundError:
+            pass
```

Comment:
Alternatively, so you don't have to use try/except, using `importlib.util.find_spec(name)` would work, as it returns `None` if it doesn't exist, and the spec if it does. No strong opinion though

Reply:
Thanks for the hint, I didn't know this possibility. I think I'll keep using the `try/except` here just because it works fine but good to know